src/command_modules/azure-cli-network/azure/cli/command_modules/network/_format.py

```diff
@@ -120,6 +120,24 @@ def transform_nsg_create_output(result):
     return {'NewNSG': result.result()}
 
 
+def transform_nsg_rule_table_output(result):
+    item = OrderedDict()
+    item['Name'] = result['name']
+    item['ResourceGroup'] = result['resourceGroup']
+    item['Priority'] = result['priority']
+    item['SourcePortRanges'] = result['sourcePortRange'] or ' '.join(result['sourcePortRanges'])
+    item['SourceAddressPrefixes'] = result['sourceAddressPrefix'] or ' '.join(result['sourceAddressPrefixes'])
+    item['SourceASG'] = result['sourceApplicationSecurityGroups'] or 'None'
```

Comment:
I assume returning a string `'None'` is the pattern for output, instead of a plain `None` to be serialized as `null` eventually

Reply:
I think i got it now, that using the string `'None'` would preventing the column from missing if when the value is `None`